### PR TITLE
Fixes regarding hive partitioning

### PR DESCRIPTION
**Pull request by yariks5s**
Needed for https://github.com/ClickHouse/ClickHouse/pull/71636

### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Implements fix, needed to enable hive partitioning by default

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit 2fab9497aa22b8c7de2ac069e6b6e202d6e44079 with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/73479/2fab9497aa22b8c7de2ac069e6b6e202d6e44079/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/73479/2fab9497aa22b8c7de2ac069e6b6e202d6e44079/style_check.html">❌ failure</a></td></tr>
<tbody>
</table>

File: src/Storages/VirtualColumnUtils.cpp [hunk 1]
```diff
@@ -170,13 +170,13 @@ VirtualColumnsDescription getVirtualsForFileLikeStorage(ColumnsDescription & sto
...
 
-    auto add_virtual = [&](const NameAndTypePair & pair)
+    auto add_virtual = [&](const NameAndTypePair & pair, bool prefer_virtual_column) /// By using prefer_virtual_column we define whether we will overwrite the storage column with the virtual one
     {
         const auto & name = pair.getNameInStorage();
         const auto & type = pair.getTypeInStorage();
         if (storage_columns.has(name))
         {
-            if (!context->getSettingsRef()[Setting::use_hive_partitioning])
+            if (prefer_virtual_column)
```
Comment (Avogar) [inline src/Storages/VirtualColumnUtils.cpp:179]: Shouldn't it be `!prefer_virtual_column`?

Comment (yariks5s) [reply src/Storages/VirtualColumnUtils.cpp:179]: Yes

**Review by Avogar — APPROVED** [review]

**Comment by yariks5s** [comment]
test_refreshable_mat_view/test.py::test_real_wait_refresh[to_clause0-False-False] - broken on master

**Pull request merged**